### DISCUSSION TITLE
Shairport-sync as a source

### DISCUSSION THREAD

**Discussion by gearhead** [mikebrady/shairport-sync#1864] [Ideas]
We all know about Sonos. Some might know about snapcast. Both of these have multi room capability where they can play from a single source to multiple devices and remain in sync. I've also used an npm package which allows airport style output to multiple rpis running shairport-sync and all remain in sync. 

Now for the idea. I have no idea if it is possible or how hard it might be, but could shairport-sync act as a source where a file or steam is played on the device in question (using MPD, or VLC for example). Then we could select one of the shairport audio outputs and one or more networked devices running shairport-sync as sinks? How possible is this? 

Gearhead

**Comment by mikebrady** [comment]
Thanks for the suggestion. It would be a pretty major piece of work to get Shairport Sync to act as a source, but have you looked at [OwnTone](https://owntone.github.io/owntone-server/)?

**Comment by gearhead** [comment]
Mike,

Thanks for the suggestion. We are already looking into it and it looks promising.

I am curious about airplay 2. We noticed that though our binary was built with the flags, we were not getting advertised as Air Play 2 in OwnTone. We did have one flag enabled  `--with-pkg-config ` and am not sure if it is needed or not. We did have to add a line to the shairport-sync.conf to get it to advertise as AirPlay 2 `regtype = "_airplay._tcp";`. Once we uncommented this, it advertised as AirPlay2. Is that the only setting in the config that needs to be set? 

Keith

**Comment by mikebrady** [comment]
OwnTone’s version of AirPlay 2 uses the NTP timing protocol, which isn’t widely used, and isn’t presently used by Apple’s own apps, so Shairport Sync does not implement it.

**Comment by gearhead** [comment]
Mike,
Thanks for the comment. It leads me to a few more questions: 

1. is the `regtype` config line the only one needed to enable AirPlay2 in shairport-sync (assuming the binary is built with this capability)? 
2. Does owntone not sync its output to multiple speakers running shairport-sync?
3. Does owntone ignore nqptp?

**Reply by mikebrady** [reply]
> Mike, Thanks for the comment. It leads me to a few more questions:
> 
> 1. is the `regtype` config line the only one needed to enable AirPlay2 in shairport-sync (assuming the binary is built with this capability)?

Not sure what this means. To be recognised as an AirPlay 2 device, the regtype must be `_airplay._tcp`.

> 2. Does owntone not sync its output to multiple speakers running shairport-sync?

OwnTone normally uses the classic AirPlay (aka "AirPlay 1") protocol and can synchronise multiple speakers, including Shairport Sync devices running in Classic Airplay mode.

> 3. Does owntone ignore nqptp?

Yes -- NQPTP uses some PTP-style signals, and OwnTone does not use these.
